draft-ietf-tls-dtls-connection-id.md

```diff
@@ -486,7 +497,11 @@ that is malicious. This attack is of concern when there is a large asymmetry
 of request/response message sizes. 
 
 Additionally, an attacker able to observe the data traffic exchanged between 
-two DTLS peers is able to replay datagrams with modified IP address/port numbers. 
+two DTLS peers is able to modify IP address/port numbers without causing the
+packet to be rejected by the recipient.  When the optional DTLS replay
+detection is not in use, such an attacker can also replay the observed packets,
```

Comment:
I'm not sure. 
For me "replay protection" requires more than the "newer filter". But the "newer filter" would already provide a level of protection, which is comparable with the "replay protection".

Reply:
The replay protection has to handle out-of-order delivery, usually with a sliding window that tracks whether the most recent N packets have been seen or not.  I'm not sure I understand your question here, though.

Reply:
Try to precise my saying:

The general issue of "replay" is **not** introduced by CID. The general protection against such "replay" is already in RFC6347. I would prefer, if this draft focus on the changes and **new** threats caused by CID and not repeat the ones, which apply also without CID.

Reply:
Let me add:

> is able to modify IP address/port numbers without causing the packet to be rejected by the recipient.

That seems to be generally true, even without CID. Otherwise (s)NATs  will not work.
What is introduced with CID, is that a **different** modification (different from such a potential modification during the handshake), may be applied. CID is especially made for that scenario. And there are no means to precise distinguish such assumed changes from those, which are applied by an attacker. There may be some scenarios considered to be more likely as an attack than others, but that are more assumptions.

Reply:
> Additionally, an attacker able to observe the data traffic exchanged between two DTLS peers is able to replay datagrams with modified IP address/port numbers.

I understand two issues here:
- message replay (reprocess the received data). CID doesn't change that, so [RFC6347 Anti-Replay](https://tools.ietf.org/html/rfc6347#section-4.1.2.6) is still valid.
- address/port change. The effect depends on the [peer-address-update](https://github.com/tlswg/dtls-conn-id/blob/master/draft-ietf-tls-dtls-connection-id.md#peer-address-update-peer-address-update).

The only change I see, that it may be considered to be easier for an observer to replay the message with the **own address** compared to replay it with a spoofed the original address as source address.

Reply:
From the discussions I wonder whether we should make this text change. It looks like it confuses more than it clarifies.